ACPs/83-dynamic-multidimensional-fees/README.md

```diff
@@ -47,24 +48,40 @@ $$base \  fee = \sum_{i=0}^3 r_i \times u_i$$
 ### Dynamic scheme components
 
 Fee rates are updated in time, to allow a fee increase when network is getting congested. Each new block is a potential source of congestion, as its transactions carry complexity that each validator must process to verify and eventually accept the block. The more complexity carries a block, and the more rapidly blocks are produced, the higher the congestion.  
-We seek a scheme that rapidly increases the fees when blocks complexity goes above a defined threshold and that equally rapidly decreases the fees once complexity goes down (because blocks carry less/simpler transactions, or because they are produced more slowly).
 
-Suppose that a block $B$ is the current chain tip. $B$ has the following features:
+We seek a scheme that rapidly increases the fees when blocks complexity goes above a defined threshold and that equally rapidly decreases the fees once complexity goes down (because blocks carry less/simpler transactions, or because they are produced more slowly). We define the desired threshold as a *target complexity rate* $T_i$: we would want to process every second a block whose cumulated complexity is $T_i$. Any complexity more than that causes some congestion that we want to penalize via fees; any congestion below $T_i$ causes the network to be under utilized.
+
+In order to update fees rates we will track, per each block and each fee dimension, a parameter called cumulated excess complexity. Fee rates applied to a block will be defined in terms of cumulated excess complexity as we show in the following.
+
+Let's define cumulated excess complexity first.
+
+Suppose that a block $B_t$ is the current chain tip. $B_t$ has the following features:
 
 - $t$ is its timestamp.
-- $B_i$ is the cumulated complexity of all its transactions along the fee dimension $i$.
-- $r_{i,t}$ is the fee rate applied to all of its transactions along the fee dimension $i$.
+- $Excess_{i,t}$ is the cumulated excess complexity along fee dimension $i$.
+
+Say a new block $B_{t + \Delta T}$ is accepted on top of $B$, with the following features:
+
+- $t + \Delta T$ is its timestamp
+- $B_i$ is its complexity.
+
+Then cumulated excess complexity once $B_{t + \Delta T}$ is accepted is updated as follow:
+
+$$Excess_{i,t + \Delta T} = max\large(0, Excess_{i,t} - T_i \times \Delta T\large) + B_i$$
```

Comment:
Could we re-organize this so that the formula goes first?

Since each term is fairly self-explanatory, I think this will make it more readable.

Reply:
Indeed, done